cfme/containers/provider/openshift.py

```diff
@@ -137,6 +158,23 @@ def __init__(
             endpoints=endpoints,
             appliance=appliance)
 
+    def create(self, **kwargs):
+
+        # Enable alerts collection before adding the provider to avoid missing active
+        # alert after adding the provider
+        # For more info: https://bugzilla.redhat.com/show_bug.cgi?id=1514950
+        if getattr(self, "alerts_type", None) == "Prometheus":
```

Comment:
**Required**

I believe this _has_ to be version picked, and even if it will work without failure in assigning the profiles, prometheus is only available in MIQ Gaprindashvili / CFME 5.9+.

The way that OpenshiftProvider.from_config() is implemented, the provider object will have an alerts type set if its in the yaml, regardless of version.

Reply:
Is this really necessary? 
you want me to change it to something like that?

``` python
getattr(self, "alerts_type") == "Prometheus" 
```

Reply:
I believe so, yes. At a minimum it keeps configuration consistent, where you would only assign enterprise profiles for a 5.9/Gaprindashvili appliance when configuring Prometheus.  Discussed with @jawatts a bit too.

The getattr() with a default use was correct, as that prevents an AttributeException - but I think since that attribute is included in the constructor with a default of None, you don't have to wrap that.

```python
if self.alerts_type.lower() == PrometheusEndpoint.name and self.appliance.version >= '5.9':
```

Reply:
I don't think that ignoring yaml corruption is right, think that if someone forgot to add this field to the YAML the entire test suite will fail, and will not run for a period of time (until the yaml be fixed and merged to the yamls repo)

I do agree that MIQ version check has to be added to this if statement

Reply:
I'm not concerned with the yaml being incorrect here - Its that a provider in the yaml with this prometheus endpoint would have these alert profiles configured, regardless of what version of the appliance is being tested.  The two OCP providers without prometheus don't have that endpoint configured, and their alerts_type will be None.

Reply:
@mshriver I thought about the 5,9 version validation and I think it needless because Prometheus is available only in version 5.9.

According to the direction of thought presented above, I have to trust the user

Reply:
@shalomnaim1 I don't think you're quite following. This isn't about 'trusting the user', and this isn't optional.  This will _not work_ with CFME/MIQ versions that don't support Prometheus, if the provider is fetched from a yaml configuration.

http://pastebin.test.redhat.com/560050